src/Workspaces/CSharp/Portable/Extensions/SyntaxTriviaListExtensions.cs

```diff
@@ -47,6 +50,37 @@ public static IEnumerable<SyntaxTrivia> SkipInitialWhitespace(this SyntaxTriviaL
             return triviaList.SkipWhile(t => t.Kind() == SyntaxKind.WhitespaceTrivia);
         }
 
+        private static ImmutableArray<ImmutableArray<SyntaxTrivia>> BreakIntoLines(this SyntaxTriviaList triviaList)
+        {
+            var result = ImmutableArray.Create<ImmutableArray<SyntaxTrivia>>();
+            var currentLine = ImmutableArray.Create<SyntaxTrivia>();
+            foreach (var trivia in triviaList)
+            {
+                currentLine = currentLine.Add(trivia);
+                if (trivia.Kind() == SyntaxKind.EndOfLineTrivia)
+                {
+                    result = result.Add(currentLine);
```

Comment:
this is `O(n^2)`.  Instead, use ArrayBuilder.GetInstance() along with `ToImmutableAndFree()`